content/pages/why.md

```diff
@@ -0,0 +1,89 @@
+* zero trust towards the server is needed,
+* no data is stored on the server,
+* servers can be self-hosted,
+* it doesn't fall apart without SSL/TLS, and
+* it's not limited to WebRTC and ORTC.
+
+# Long
+
+We'll take WebRTC signalling solutions as an example to describe signalling 
+types and their security issues.
+
+Most existing applications that utilise WebRTC claim that their users are fully 
+end-to-end encrypted because WebRTC provides end-to-end encryption. However, 
+this is not true. Because if the signalling is not end-to-end encrypted, any 
+two peers may connect to one another but it is not guaranteed that one peer 
+connects to another peer it actually wanted to establish the connection with. A 
+man-in-the-middle attack is still very possible. Thus, signalling information 
+authentication and integrity is the weakest link of WebRTC's end-to-end 
+encryption claim.
+
+But the man-in-the-middle attack isn't the only problem. WebRTC's signalling 
+data also contains information about the peer's IP addresses which may leak 
+information about its location. It possibly contains information that can be 
+used to conclude browser vendor and version, codecs which may lead to installed 
+libraries and their versions, and so on and so forth. Thus, we do not want 
+anyone else but the other peer we want to communicate with to have access to 
+this data. The only way of doing this is by using end-to-end encryption.
+
+Now that we know why it's important to end-to-end encrypt the signalling 
+messages, we'll go through existing signalling solutions and their problems.
+
+## Transport Encryption Solutions
+
+There are lots and lots of available hosted signalling solutions for WebRTC. 
+Most of them provide transport encryption and thus require the end user of the 
+WebRTC application to trust a third party that relays signalling messages from 
+one peer to another before they can communicate directly. For example, 
+[Firebase][firebase] offers such a solution. You will have to trust that they 
+have no security issues regarding their SSL/TLS settings and certificates. 
+Furthermore, providers such as Firebase have full access to the exchanged 
+signalling messages in plaintext and are technically able to modify them. While 
+you may choose to trust Firebase to not modify or use signalling data in any 
+way, they still have obligations to follow national laws and might be compelled 
+to hand out the end user's signalling data.
+
+## Self-Hosted Transport Encryption Solutions
+
+There are also plenty of transport encrypted signalling solutions that allow 
+you to host them on your own server, such as [PeerJS][peer-js]. Although this 
+is a better solution than trusting a third party, the end user still has to 
+trust you to not modify signalling messages and not have any security issues on 
+your severs. Even though we usually don't want to admit it, secretly we all 
+know that certificate and private key handling isn't easy. Moreover, you're 
```

Comment:
SaltyRTC still requires some key handling, right? What's the worst that could happen due to mis-handling of keys?